relay/sources/relayd/src/api/shared_folder.rs

```diff
@@ -53,46 +51,33 @@ fn default_hash() -> String {
 pub fn head(
     params: SharedFolderParams,
     // Relative path
-    file: &Path,
+    file: PathBuf,
     job_config: Arc<JobConfig>,
-) -> Result<warp::reply::WithStatus<String>, Error> {
-    let file_path = job_config.cfg.shared_folder.path.join(file);
+) -> impl Future<Item = StatusCode, Error = Error> + Send {
+    let file_path = job_config.cfg.shared_folder.path.join(&file);
     debug!(
         "Received request for {:#} ({:#} locally) with the following parameters: {:?}",
         file.display(),
         file_path.display(),
         params
     );
-    let hash_type = HashType::from_str(&params.hash_type)?;
 
-    Ok(if file_path.exists() {
-        if params.hash.is_empty() {
-            reply::with_status(
-                format!("{} exists but hash is empty", file.display()),
-                StatusCode::OK,
-            )
-        } else {
-            let actual_hash = hash_type.hash(&fs::read(file_path)?);
-            debug!("{} {} hash is {}", file.display(), hash_type, params.hash);
-            if params.hash == actual_hash {
-                reply::with_status(
-                    format!("{} exists and has same hash", file.display()),
-                    StatusCode::NOT_MODIFIED,
-                )
-            } else {
-                reply::with_status(
-                    format!(
-                        "{:#} exists but its hash is which is different",
-                        file.display()
-                    ),
-                    StatusCode::OK,
-                )
+    futures::future::result(HashType::from_str(&params.hash_type)).and_then(move |hash_type| {
```

Comment:
there's no checking anymore on parameters, is it expected ?

Reply:
We lost a bit of log precision indeed! I'm fixing this.